placement/algo/sharded_helper.go

```diff
@@ -88,15 +88,15 @@ type PlacementHelper interface {
 	// Instances returns the list of instances managed by the PlacementHelper.
 	Instances() []placement.Instance
 
-	// HasRackConflict checks if the rack constraint is violated when moving the shard to the target rack.
-	HasRackConflict(shard uint32, from placement.Instance, toRack string) bool
+	// CanMove checks if the shard can be moved from the instance to the target isolation group.
+	CanMove(shard uint32, from placement.Instance, isolationGroup string) bool
```

Comment:
Isn't this method the inverse of `CanMove`?  It seems to only return true if you cannot move it (i.e. there is a conflict).

Would a better name be `CannotPlaceShardInIsolationGroupFromInstance(...)`?  Also using parameters `...(shard, isolationGroup, placement.Instance)` might make more sense to match the order of things described by the name.

Reply:
Yeah I think making it `CanMoveShard` is probably more clear, I'm going to reverse the implementation and usage instead